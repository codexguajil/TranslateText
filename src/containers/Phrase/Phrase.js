import React, { Component } from 'react'
import { connect } from 'react-redux'

export class Phrase extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  addFave = () => {
    
  }

  deletePhrase = () => {

  }

  render() {
    return (
      <div onClick={this.addFave}
           className='text-box'
      >{this.props.translatedText}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  translations: state.translations
})

const mapDispatchToProps = (dispatch) => ({

})

export default connect(mapStateToProps)(Phrase)