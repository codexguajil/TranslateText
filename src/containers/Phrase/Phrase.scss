.text-box {
  background: #FFF;
  border: 1px dotted #FFF;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
  color: black;
  font-size: 26px;
  height: 80px;
  margin: 30px;
  opacity: 0.9;
  padding: 8px;
  text-align: left;
  font-weight: 300;
  width: 29%;
  &:hover {
    background: #ffefc1;
    cursor: pointer;
    font-weight: normal;
    transition: background-color .5s ease;
  }
}