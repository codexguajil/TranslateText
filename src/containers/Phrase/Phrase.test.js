import {Phrase} from './Phrase'
import React from 'react'
import {shallow} from 'enzyme'

describe('Phrase', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(
      <Phrase />
    )

    expect(wrapper).toMatchSnapshot()
  })
})