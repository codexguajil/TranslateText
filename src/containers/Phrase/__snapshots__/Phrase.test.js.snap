// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phrase should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Phrase />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="text-link"
          to="/translations/undefined"
        >
          <div
            className="text-box"
          />
        </Link>,
        <button
          className="delete"
          onClick={[Function]}
        >
          delete
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="text-box"
          />,
          "className": "text-link",
          "to": "/translations/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "text-box",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "delete",
          "className": "delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="text-link"
            to="/translations/undefined"
          >
            <div
              className="text-box"
            />
          </Link>,
          <button
            className="delete"
            onClick={[Function]}
          >
            delete
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="text-box"
            />,
            "className": "text-link",
            "to": "/translations/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "text-box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "delete",
            "className": "delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
